### DISCUSSION TITLE
Animating scale, rotation, and scroll at the same time

### DISCUSSION THREAD

**Discussion by TimPushkin** [peterLaurence/MapCompose#82] [Ideas]
I would like to be able to scale, rotate, and scroll the map at the same time. My use case is that I am developing a navigation app where I have a marker which is the current user position with a cone of sight, and when the position changes I want to re-center on this marker and at the same time rotate the map so the cone of sight is directed at the top of the screen again.

Currently the closest easy thing I can do with the `MapState` API is to sequentially call `centerOnMarker()` and then `rotateTo()`, but I cannot call them concurrently as I would like to because they use the same `apiAnimatable` property of `ZoomPanRotateState`. It seems like I can also animate rotation myself, but this will involve reimplementing some of `rotateTo()` functionality.

Since some APIs already allow scroll and scale animations to be performed at the same time, it would be nice to also have rotation there. For me personally, `centerOnMarker()` is particularly interesting.

**Comment by p-lr** [comment]
This makes sense, I'll have a look. Thanks for the suggestion.

**Comment by p-lr** [comment]
I've pushed the branch `center-and-rotate`, in which the `centerOnMarker()` has a new overload:
```kotlin
suspend fun MapState.centerOnMarker(
    id: String,
    destScale: Float,
    destAngle: AngleDegree,
    animationSpec: AnimationSpec<Float> = SpringSpec(stiffness = Spring.StiffnessLow)
)
```

The "CenteringOnMarker" demo has been updated to test this new api.
I've also updated the Compose BOM to 2023.03.00 and the compiler for the latest stable version, which *required an update of Kotlin to 1.8.10*. In my custom app, I had to update the BOM, Kotlin and the Compose compiler too (to avoid runtime issue).
Let me know if that works for you.

**Reply by TimPushkin** [reply]
Tried the demo and it looks like what I need, thanks!

**Reply by p-lr** [reply]
The `2.5.0` is published. You'll have to use the BOM 2023.03.00 and Kotlin 1.8.10 in your app.
